scripts/src/services/users.ts

```diff
@@ -46,3 +55,63 @@ export function validateJWT(jwt: string): {userId: string, appId: string} {
 		appId: decoded.payload.iss,
 	};
 }
+
+export async function getOrCreateUserCredentials(
+	jwt: string,
+	walletAddress: string,
+	deviceId: string): Promise<AuthToken> {
+
+	const { userId , appId } = validateJWT(jwt); // throws if JWT not valid // XXX test this case
+	let user = await db.User.findOne( { appId, appUserId: userId });
+	if (!user) {
+		// new user
+		user = new db.User();
```

Comment:
This is an ugly way of doing it.
You can do this:
```
user = Object.assign(new db.User(), {
    appId, userId, walletAddress
});
```
And should also place that in the `User` ctor

Reply:
yes, It's a copy from their documentation. Usually I wrap this kind of creation in a static `create` function